src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -76,7 +76,7 @@ class CollectorService(
         streamObservable = mqSubject.flatMapIterable(MessageBatch::getMessagesList)
                 .groupBy { message ->
                     message.metadata.id.run {
-                        SessionKey(connectionId.sessionAlias, direction)
+                        SessionKey(message.parentEventId.bookName, connectionId.sessionAlias, direction)
```

Comment:
```suggestion
                        SessionKey(bookName, connectionId.sessionAlias, direction)
```
We should collect messages by their metadata.

Reply:
done